app/src/lib/stores/app-store.ts

```diff
@@ -2357,8 +2358,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
   private async matchGitHubRepository(
     repository: Repository
   ): Promise<IMatchedGitHubRepository | null> {
-    const gitStore = this.getGitStore(repository)
-    const remote = gitStore.defaultRemote
+    const remote = await getDefaultRemote(repository)
```

Comment:
I wanted to get away from sprinkling `getDefaultRemote` (which requires spawning Git to enumerate all remotes) around the place as part of #5399 because this value is never cached, and the other code paths around refreshing a repository already do a `gitStore.loadRemotes()` to load and cache the default remote:

 - `AppStore._refreshRepository()`
 - `AppStore._publishRepository()`

Are we able to ensure we're in a good state before doing this first refresh (like the other code paths) by loading the remotes? This patch seems to work fine for me:

```diff
diff --git a/app/src/lib/stores/app-store.ts b/app/src/lib/stores/app-store.ts
index a8d513107..fe9ac69e0 100644
--- a/app/src/lib/stores/app-store.ts
+++ b/app/src/lib/stores/app-store.ts
@@ -160,7 +160,6 @@ import { findAccountForRemoteURL } from '../find-account'
 import { inferLastPushForRepository } from '../infer-last-push-for-repository'
 import { MergeResultKind } from '../../models/merge'
 import { promiseWithMinimumTimeout } from '../promise'
-import { getDefaultRemote } from '../../lib/git/remote'
 
 /**
  * Enum used by fetch to determine if
@@ -2358,7 +2357,8 @@ export class AppStore extends TypedBaseStore<IAppState> {
   private async matchGitHubRepository(
     repository: Repository
   ): Promise<IMatchedGitHubRepository | null> {
-    const remote = await getDefaultRemote(repository)
+    const gitStore = this.getGitStore(repository)
+    const remote = gitStore.defaultRemote
     return remote !== null
       ? matchGitHubRepository(this.accounts, remote.url)
       : null
@@ -3438,6 +3438,12 @@ export class AppStore extends TypedBaseStore<IAppState> {
         const addedRepo = await this.repositoriesStore.addRepository(
           validatedPath
         )
+
+        // initialize the remotes for this new repository to ensure it can fetch
+        // it's GitHub-related details using the GitHub API (if applicable)
+        const gitStore = this.getGitStore(addedRepo)
+        await gitStore.loadRemotes()
+
         const [refreshedRepo, usingLFS] = await Promise.all([
           this.repositoryWithRefreshedGitHubRepository(addedRepo),
           this.isUsingLFS(addedRepo),
```

Reply:
We could exclude it from [`app/src/lib/git/index.ts`](https://github.com/desktop/desktop/blob/master/app/src/lib/git/index.ts) so that we can write tests against it, but callers in the app are not using it.

https://github.com/desktop/desktop/blob/741a5534bcaed63a5b6c5637feb59bd64bdc27c3/app/src/lib/git/index.ts#L19

Reply:
Alternatively, if those tests are not valuable and there's no public callers we should just :fire: the code and ensure we have good test coverage elsewhere

Reply:
lets continue discussion at #5626